File: src/javascript/user/message/list-messages.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

const client = new FaxCoreClient();

const result = await client.post('/api/message/list', {
  folderName: process.env.FAXCORE_FOLDER || 'inbox',
  startDate: process.env.FAXCORE_START_DATE || '20000101',
  endDate: process.env.FAXCORE_END_DATE || '20991231',
  isRead: process.env.FAXCORE_IS_READ || 'all',
  isDownloaded: process.env.FAXCORE_IS_DOWNLOADED || 'all',
  sortDescending: true,
  pagination: {
    page: Number(process.env.FAXCORE_PAGE || 1),
    maxResult: Number(process.env.FAXCORE_MAX_RESULT || 25)
  }
});

printResult('Message list', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/Message/ListMessages.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Message
{
    internal static class ListMessages
    {
        public static Task<string> RunAsync(FaxCoreClient client)
        {
            var request = new
            {
                folderName = Environment.GetEnvironmentVariable("FAXCORE_FOLDER") ?? "inbox",
                startDate = Environment.GetEnvironmentVariable("FAXCORE_START_DATE") ?? "20000101",
                endDate = Environment.GetEnvironmentVariable("FAXCORE_END_DATE") ?? "20991231",
                isRead = Environment.GetEnvironmentVariable("FAXCORE_IS_READ") ?? "all",
                isDownloaded = Environment.GetEnvironmentVariable("FAXCORE_IS_DOWNLOADED") ?? "all",
                sortDescending = true,
                pagination = new
                {
                    page = 1,
                    maxResult = 25
                }
            };

            return client.PostAsync("/api/message/list", request);
        }
    }
}
